chunky/src/java/se/llbit/chunky/renderer/RenderWorker.java

```diff
@@ -133,10 +135,7 @@ private void work(RenderTile tile) {
             sg += ray.color.y;
             sb += ray.color.z;
           }
-          double sinv = 1.0 / (scene.spp + manager.sppPerPass);
-          samples[offset + 0] = (samples[offset + 0] * scene.spp + sr) * sinv;
-          samples[offset + 1] = (samples[offset + 1] * scene.spp + sg) * sinv;
-          samples[offset + 2] = (samples[offset + 2] * scene.spp + sb) * sinv;
+          samples.addSample(x, y, scene.spp, manager.sppPerPass, sr, sg, sb);
```

Comment:
Maybe we could change this so that the `SampleBuffer` "knows" its spp (or later spp at the specific pixel) so that only `manager.sppPerPass` needs to be passed here as a "weight".